SecondClasses/rev_polish_notation/src/rev_polish_notation.erl

```diff
@@ -2,6 +2,28 @@
 
 -export([evaluate/1]).
 
+evaluate(List)->
+    R = lists:foldl(fun next/2, [], List),
+    lists:last(R).
 
-evaluate(_) ->
-    1.
\ No newline at end of file
+next(Elem, Stack) when is_integer(Elem) ->
+    [Elem | Stack];
+next(Elem, Stack) when is_function(Elem)->
```

Comment:
How about:
```erlang
next(Elem, [H | T]) when is_function(Elem, 1) ->
    ...
next(Elem, [H1, H2 | T]) when is_function(Elem, 2) ->
    ...
next(Elem, Stack) when is_number(Elem) ->
    [Elem | Stack];
next(Elem, Stack) -> {error, "reason"}.
```
?
what if I pass floats like this `[0.5, 0.3, erlang:'+'/2]`?
cosinder using http://erlang.org/doc/man/erlang.html#is_number-1 instead od is_integer/1 ;)